### PR TITLE
chore: update DF to 981a9bbc229288da5293a4a5f0478bdf5ac358df

### PR DESCRIPTION
**Pull request by crepererum**
Ref #6581.

### CODE REVIEW

File: query_tests/cases/in/dedup_and_predicates_parquet_ingester.expected [hunk 1]
```diff
@@ -8,24 +8,24 @@
...
-|               |         DeduplicateExec: [tag@2 ASC,time@3 ASC]                                                                                                                                                  |
-|               |           SortPreservingMergeExec: [tag@2 ASC,time@3 ASC]                                                                                                                                        |
-|               |             UnionExec                                                                                                                                                                            |
-|               |               ParquetExec: limit=None, partitions={1 group: [[1/1/1/1/00000000-0000-0000-0000-000000000000.parquet]]}, output_ordering=[tag@2 ASC, time@3 ASC], projection=[bar, foo, tag, time] |
-|               |               SortExec: [tag@2 ASC,time@3 ASC]                                                                                                                                                   |
```
Comment (crepererum) [inline query_tests/cases/in/dedup_and_predicates_parquet_ingester.expected:25]: This seems wrong: why is it bubbling up the sort here? The parquet file data is sorted, no need to sort it again.

File: query_tests/cases/in/several_chunks.expected [hunk 2]
```diff
@@ -28,8 +28,7 @@
...
 |               |           ParquetExec: limit=None, partitions={1 group: [[1/1/1/1/00000000-0000-0000-0000-000000000002.parquet]]}, output_ordering=[city@0 ASC, state@2 ASC, time@4 ASC], projection=[city, other_temp, state, temp, time] |
 |               |     DeduplicateExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                                                                                                                                   |
-|               |       SortExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                                                                                                                                        |
-|               |         RecordBatchesExec: batches_groups=1 batches=1                                                                                                                                                                      |
+|               |       RecordBatchesExec: batches_groups=1 batches=1                                                                                                                                                                        |
```
Comment (crepererum) [inline query_tests/cases/in/several_chunks.expected:31]: This also seems broken: IIRC we treat the record batch exec as unsorted, so it MUST be sorted prior to de-dup.

File: query_tests/cases/in/two_chunks.expected [hunk 3]
```diff
@@ -34,18 +34,18 @@
...
+|               |   ProjectionExec: expr=[other_temp@1 as other_temp, temp@3 as temp, time@4 as time]                                                                                                                                          |
+|               |     DeduplicateExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                                                                                                                                     |
+|               |       SortPreservingMergeExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                                                                                                                           |
+|               |         SortExec: [city@0 ASC,state@2 ASC,time@4 ASC]                                                                                                                                                                        |
```
Comment (alamb) [inline query_tests/cases/in/two_chunks.expected:46]: THis is a very bad change -- it means the output of the already sorted parquet files will be resorted

See update on  https://github.com/apache/arrow-datafusion/commit/8ec511e51cdfdef8e3f79116076fbb962f53f887

Comment (alamb) [reply query_tests/cases/in/two_chunks.expected:46]: Note I think I have fixed the upstream issue: https://github.com/apache/arrow-datafusion/pull/4885

Once I verify that works then we can start maybe doing some incremental updates of IOx

**Comment by crepererum** [comment]
Closing since @alamb is now working on an actual upgrade incl. the upstream fixes.

**Pull request closed without merging**